blackbox/docs/config/secure.rst

```diff
@@ -0,0 +1,73 @@
+.. highlight:: sh
+
+.. _conf-secure-settings:
+
+===============
+Secure Settings
+===============
+
+Some settings are sensitive as they might store passwords, access keys or 
+other similar information. This type of data needs more protection than just 
+filesystem permissions and even more, should not be available in plain text.
+
+For these instances, we provide a keystore and the ``crate-keystore`` tool to 
+manage the settings in the keystore.
+
+.. NOTE::
+
+    Changes to the keystore take affect only after restarting CrateDB
+
+.. NOTE::
+
+    The CrateDB keystore currently provides obfuscation, but in the future 
+    password protection will be added.
+
+.. NOTE::
+
+    Not all settings are read from the keystore. Consult the documentation for 
+    each setting to see if it is secure setting.
+
+
+Creating the keystore
+---------------------
+To create the crate.keystore, use the ``create`` command:
+
+.. code-block:: sh
+
+    sh$ ./bin/crate-keystore create
+
+The file ``crate.keystore`` will be created alongside ``crate.yml``.
+
+Listing settings in the keystore
+--------------------------------
+To list all the settings in the keystore, use the ``list`` command:
+
+.. code-block:: sh
+
+    bsh$ ./bin/crate-keystore list
+
+Adding string settings
+----------------------
+Sensitive string settings, like authentication credentials for cloud plugins, 
+can be added using the ``add`` command:
+
+
+.. code-block:: sh
+
+    sh$ ./bin/crate-keystore add setting.name
+
+The script will then prompt for the value of the setting. To pass the value 
+through stdin, use the ``--stdin`` flag:
```

Comment:
is there also an option for passing it as an argument? i.e. `add setting.name VALUE`

Reply:
no , I don't think so

Reply:
okay! odd! but ok!

Reply:
its not so odd as one should not pass in sensitive values via cmd in general as these values would be then stored inside the bash history